docs/src/tools.rst

```diff
@@ -124,13 +124,13 @@ with :github-source:`tools/integreat-cms-cli`::
 Code Quality
 ============
 
-Automatically sort python import statements with :github-source:`tools/isort.sh`::
+(Deprecated) Automatically apply our python style with :github-source:`tools/black.sh`::
```

Comment:
```suggestion
Automatically apply our python style with :github-source:`tools/black.sh`::
```

I'd argue it is not deprecated yet, since using `ruff format` leads to a very lage diff right now.

Reply:
You might want to consider `ruff format` as a highly compatible but much faster replacement for `black`.
* https://docs.astral.sh/ruff/formatter/#black-compatibility

Reply:
The diff is largely due to currently wrapped lines (black's default line length is 88) being put on a single line (pyproject.toml configures line lengths of 249 for ruff, apparently due to differences in the way comment strings are habndled, at least so I've been told)

Reply:
https://docs.astral.sh/ruff/formatter/black/#trailing-end-of-line-comments

Reply:
From that article:

> We agree that Ruff's formatting (that matches Black's 23) is hard to read and needs improvement. But we aren't convinced that parenthesizing long nested expressions is the best solution, especially when considering expression formatting holistically. That's why we want to defer the decision until we've explored alternative nested expression formatting styles.

However, setting 
```toml
[tool.ruff]
line-length = 88
```

indeed does introduce only a handful of changes - @timobrembeck, did I misunderstand the reason for setting it to `279`? Because just from this it looks like we could indeed shorten this.

Reply:
I guess this is the corresponding issue on ruff's side: https://github.com/astral-sh/ruff/issues/9926

Reply:
Use `ruff format` instead of `black`?  Highly compatible and much faster.

Reply:
+1 for ruff format @timobrembeck that throws 0 errors and only reformats 15 files

Reply:
Awesome...  Use `ruff check --select=PL` instead of `pylint`?

Reply:
@cclauss, this seems like a very good idea to me in general, but Ruff does not yet support all Pylint rules. Therefore, I'd prefer to wait a bit and ensure a smooth migration later on.

Reply:
@charludo ruff formatting does work on my local setup as well, but it fails within the pre-commit hook and on circleci as well. I guess we have to wait with the formatter migration until the bug in the rule is fixed.